src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs

```diff
@@ -99,19 +102,29 @@ public static TracerProviderBuilder AddProcessor(this TracerProviderBuilder trac
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
         public static TracerProviderBuilder AddLegacySource(this TracerProviderBuilder tracerProviderBuilder, string operationName)
         {
-            if (tracerProviderBuilder is TracerProviderBuilderSdk tracerProviderBuilderSdk)
+            if (tracerProviderBuilder is TracerProviderBuilderBase tracerProviderBuilderBase)
             {
-                tracerProviderBuilderSdk.AddLegacySource(operationName);
+                tracerProviderBuilderBase.AddLegacySource(operationName);
             }
 
             return tracerProviderBuilder;
         }
 
+        /// <summary>
+        /// Run the given actions to initialize the <see cref="TracerProvider"/>.
+        /// </summary>
+        /// <param name="tracerProviderBuilder"><see cref="TracerProviderBuilder"/>.</param>
+        /// <returns><see cref="TracerProvider"/>.</returns>
         public static TracerProvider Build(this TracerProviderBuilder tracerProviderBuilder)
         {
-            if (tracerProviderBuilder is TracerProviderBuilderSdk tracerProviderBuilderSdk)
+            if (tracerProviderBuilder is IDeferredTracerProviderBuilder)
+            {
+                throw new NotSupportedException("DeferredTracerBuilder requires a ServiceProvider to build.");
```

Comment:
This can't be here because the `Build` method from the `TracerProviderBuilderHosting` class calls this method and then it blows up the build process.

Reply:
Whoops! Fixed it up by making the build on TracerProviderBuilderBase protected. I feel like we should keep this throw here to prevent anyone from accidentally using the hosting builder with the non-IServerProvider build extension?